control/server.py

```diff
@@ -30,17 +31,38 @@
 from .state import GatewayState, LocalGatewayState, OmapGatewayState, GatewayStateHandler
 from .grpc import GatewayService
 
-libc = ctypes.CDLL(ctypes.util.find_library("c"))
-PR_SET_PDEATHSIG = 1
+def wait_for_condition(condition_method: Callable[[], bool], error_message=None, timeout=None, interval=1):
+    """Use the provided condition_method predicate to await a specific condition.
+
+       If the condition evaluates to False even after the timeout, raise a TimeoutError.
+    """
+    start_time = time.time()
+
+    while True:
+        if condition_method():
+            return
 
+        if timeout is not None and time.time() - start_time >= timeout:
+            raise TimeoutError(f"Timeout: {error_message} after {timeout} seconds.")
```

Comment:
nit: with `error_message=None` (which is the default) it will not look very nice.

Reply:
I wouldn't repeat the `Timeout` part, since exception messages are always prefixed by exception names:
```py
raise TimeoutError("hello error")
TimeoutError: hello error
```
```suggestion
            raise TimeoutError(f"{error_message} (after {timeout} seconds)")
```
Example:
```py
>>> raise TimeoutError(f"{error_message} (after {timeout} seconds)")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TimeoutError: File not found (after 5 seconds)
```

Reply:
Thank you Ernesto. Removed [wait_for_condition](https://github.com/ceph/ceph-nvmeof/pull/182#discussion_r1299620078) for now per Mykola comment